--- a/IF_Trellis.py	2026-06-17 14:12:51.312855410 +0200
+++ b/IF_Trellis.py	2026-06-17 14:12:51.354653052 +0200
@@ -243,7 +243,8 @@
         del mesh_output
         torch.cuda.empty_cache()
         
-        logger.info(f"Texture image shape: {texture_image.shape}")
+        if texture_image is not None:
+            logger.info(f"Texture image shape: {texture_image.shape}")
 
         return video_path, glb_path, texture_path, wireframe_path, texture_image, wireframe_image
 
