diff -Naur appmenu-gtk-0.2.1/src/Makefile.am fixed/appmenu-gtk-0.2.1/work/appmenu-gtk-0.2.1/src/Makefile.am
--- appmenu-gtk-0.2.1/src/Makefile.am	2011-09-12 16:53:52.000000000 +0000
+++ fixed/appmenu-gtk-0.2.1/work/appmenu-gtk-0.2.1/src/Makefile.am	2011-09-12 16:54:55.000000000 +0000
@@ -13,7 +13,7 @@
 	gen-application-menu-registrar.xml.h
 
 libappmenu_la_CPPFLAGS =	\
-	-Wall -Werror		\
+	-Wall 		\
 	$(APPMENU_CFLAGS)
 
 libappmenu_la_LDFLAGS =		\
diff -Naur appmenu-gtk-0.2.1/src/Makefile.in fixed/appmenu-gtk-0.2.1/work/appmenu-gtk-0.2.1/src/Makefile.in
--- appmenu-gtk-0.2.1/src/Makefile.in	2011-09-12 16:53:52.000000000 +0000
+++ fixed/appmenu-gtk-0.2.1/work/appmenu-gtk-0.2.1/src/Makefile.in	2011-09-12 16:55:35.000000000 +0000
@@ -256,7 +256,7 @@
 	gen-application-menu-registrar.xml.h
 
 libappmenu_la_CPPFLAGS = \
-	-Wall -Werror		\
+	-Wall		\
 	$(APPMENU_CFLAGS)
 
 libappmenu_la_LDFLAGS = \
 