--- a/doc/conf.py	2017-01-17 18:30:22.000000000 +0800
+++ b/doc/conf.py	2023-03-27 12:14:47.395994501 +0800
@@ -66,12 +66,15 @@
     'sphinxcontrib.bibtex',
 ]
 
+# List of bib files
+bibtex_bibfiles = ['bibliography.bib']
+
 # Undocumented trick: if we def setup here in conf.py, it gets called just
 # like an extension's setup function.
 def setup(app):
-    app.add_javascript("show-code.js")
-    app.add_javascript("facebox.js")
-    app.add_stylesheet("facebox.css")
+    app.add_js_file("show-code.js")
+    app.add_js_file("facebox.js")
+    app.add_css_file("facebox.css")
 
 # Add any paths that contain templates here, relative to this directory.
 templates_path = ['_templates']
