--- RunBundler.sh	2014-10-25 17:26:18.000000000 +0400
+++ RunBundler.sh	2014-10-25 17:29:10.921078002 +0400
@@ -33,21 +33,21 @@
     exit;
 fi
 
-EXTRACT_FOCAL=$BASE_PATH/bin/extract_focal.pl
+EXTRACT_FOCAL=$BASE_PATH/extract_focal.pl
 
 OS=`uname -o`
 
 if [ $OS == "Cygwin" ]
 then
-    MATCHKEYS=$BASE_PATH/bin/KeyMatchFull.exe
-    BUNDLER=$BASE_PATH/bin/Bundler.exe
+    MATCHKEYS=$BASE_PATH/KeyMatchFull.exe
+    BUNDLER=$BASE_PATH/Bundler.exe
 else
-    MATCHKEYS=$BASE_PATH/bin/KeyMatchFull
-    BUNDLER=$BASE_PATH/bin/bundler
+    MATCHKEYS=$BASE_PATH/KeyMatchFull
+    BUNDLER=$BASE_PATH/bundler
 fi
 
-TO_SIFT=$BASE_PATH/bin/ToSift.sh
-TO_SIFT_LIST=$BASE_PATH/bin/ToSiftList.sh
+TO_SIFT=$BASE_PATH/ToSift.sh
+TO_SIFT_LIST=$BASE_PATH/ToSiftList.sh
 
 # Default config options
 IMAGE_DIR="."
