diff --git a/src/base/linuxthreads.cc b/src/base/linuxthreads.cc
index 19da400..c578d39 100644
--- a/src/base/linuxthreads.cc
+++ b/src/base/linuxthreads.cc
@@ -193,7 +193,7 @@ static volatile int *sig_pids, sig_num_threads, sig_proc, sig_marker;
 /* Signal handler to help us recover from dying while we are attached to
  * other threads.
  */
-static void SignalHandler(int signum, siginfo_t *si, void *data) {
+static void SignalHandler(int signum, siginfo *si, void *data) {
   if (sig_pids != NULL) {
     if (signum == SIGABRT) {
       while (sig_num_threads-- > 0) {
