diff -Naur /var/tmp/portage/media-sound/deadbeef-9999/work/deadbeef-9999/plugins/gtkui/interface.c deadbeef-9999/plugins/gtkui/interface.c
--- /var/tmp/portage/media-sound/deadbeef-9999/work/deadbeef-9999/plugins/gtkui/interface.c	2012-09-09 18:03:06.000000000 +0000
+++ deadbeef-9999/plugins/gtkui/interface.c	2012-09-09 18:07:53.000000000 +0000
@@ -198,9 +198,6 @@
   new_playlist1 = gtk_menu_item_new_with_mnemonic (_("New playlist"));
   gtk_widget_show (new_playlist1);
   gtk_container_add (GTK_CONTAINER (File_menu), new_playlist1);
-  gtk_widget_add_accelerator (new_playlist1, "activate", accel_group,
-                              GDK_n, (GdkModifierType) GDK_CONTROL_MASK,
-                              GTK_ACCEL_VISIBLE);
 
   playlist_load = gtk_menu_item_new_with_mnemonic (_("Load playlist"));
   gtk_widget_show (playlist_load);
@@ -508,9 +505,7 @@
   gtk_widget_show (stopbtn);
   gtk_box_pack_start (GTK_BOX (hbox3), stopbtn, FALSE, FALSE, 0);
   gtk_widget_set_can_focus(stopbtn, FALSE);
-  gtk_widget_add_accelerator (stopbtn, "activate", accel_group,
-                              GDK_v, (GdkModifierType) 0,
-                              GTK_ACCEL_VISIBLE);
+
   gtk_button_set_relief (GTK_BUTTON (stopbtn), GTK_RELIEF_NONE);
 
   image128 = gtk_image_new_from_stock ("gtk-media-stop", GTK_ICON_SIZE_BUTTON);
@@ -521,12 +516,7 @@
   gtk_widget_show (playbtn);
   gtk_box_pack_start (GTK_BOX (hbox3), playbtn, FALSE, FALSE, 0);
   gtk_widget_set_can_focus(playbtn, FALSE);
-  gtk_widget_add_accelerator (playbtn, "activate", accel_group,
-                              GDK_x, (GdkModifierType) 0,
-                              GTK_ACCEL_VISIBLE);
-  gtk_widget_add_accelerator (playbtn, "activate", accel_group,
-                              GDK_Return, (GdkModifierType) 0,
-                              GTK_ACCEL_VISIBLE);
+
   gtk_widget_add_accelerator (playbtn, "activate", accel_group,
                               GDK_KP_Enter, (GdkModifierType) 0,
                               GTK_ACCEL_VISIBLE);
@@ -540,12 +530,7 @@
   gtk_widget_show (pausebtn);
   gtk_box_pack_start (GTK_BOX (hbox3), pausebtn, FALSE, FALSE, 0);
   gtk_widget_set_can_focus(pausebtn, FALSE);
-  gtk_widget_add_accelerator (pausebtn, "activate", accel_group,
-                              GDK_c, (GdkModifierType) 0,
-                              GTK_ACCEL_VISIBLE);
-  gtk_widget_add_accelerator (pausebtn, "activate", accel_group,
-                              GDK_p, (GdkModifierType) GDK_CONTROL_MASK,
-                              GTK_ACCEL_VISIBLE);
+
   gtk_button_set_relief (GTK_BUTTON (pausebtn), GTK_RELIEF_NONE);
 
   image3 = gtk_image_new_from_stock ("gtk-media-pause", GTK_ICON_SIZE_BUTTON);
@@ -556,9 +541,7 @@
   gtk_widget_show (prevbtn);
   gtk_box_pack_start (GTK_BOX (hbox3), prevbtn, FALSE, FALSE, 0);
   gtk_widget_set_can_focus(prevbtn, FALSE);
-  gtk_widget_add_accelerator (prevbtn, "activate", accel_group,
-                              GDK_z, (GdkModifierType) 0,
-                              GTK_ACCEL_VISIBLE);
+
   gtk_button_set_relief (GTK_BUTTON (prevbtn), GTK_RELIEF_NONE);
 
   image4 = gtk_image_new_from_stock ("gtk-media-previous", GTK_ICON_SIZE_BUTTON);
@@ -569,9 +552,7 @@
   gtk_widget_show (nextbtn);
   gtk_box_pack_start (GTK_BOX (hbox3), nextbtn, FALSE, FALSE, 0);
   gtk_widget_set_can_focus(nextbtn, FALSE);
-  gtk_widget_add_accelerator (nextbtn, "activate", accel_group,
-                              GDK_b, (GdkModifierType) 0,
-                              GTK_ACCEL_VISIBLE);
+
   gtk_button_set_relief (GTK_BUTTON (nextbtn), GTK_RELIEF_NONE);
 
   image5 = gtk_image_new_from_stock ("gtk-media-next", GTK_ICON_SIZE_BUTTON);
