diff -Naur wavelet-denoise-0.3.1/src/Makefile wavelet-denoise-0.3.1/src/Makefile
--- wavelet-denoise-0.3.1/src/Makefile	2008-12-14 11:19:28.000000000 +0000
+++ wavelet-denoise-0.3.1/src/Makefile	2013-10-20 11:52:49.091807134 +0000
@@ -1,5 +1,5 @@
 CFLAGS = -O3 -Wall $(shell gimptool-2.0 --cflags)
-LIBS = $(shell gimptool-2.0 --libs)
+LIBS = -lm $(shell gimptool-2.0 --libs)
 PLUGIN = wavelet-denoise
 SOURCES = plugin.c colorspace.c denoise.c wavelet.c events.c interface.c
 HEADERS = plugin.h interface.h messages.h

diff -Naur wavelet-denoise-0.3.1/po/Makefile wavelet-denoise-0.3.1/po/Makefile
--- wavelet-denoise-0.3.1/po/Makefile	2008-12-20 11:23:05.000000000 +0000
+++ wavelet-denoise-0.3.1/po/Makefile	2013-10-20 11:57:19.493795762 +0000
@@ -24,7 +24,8 @@
 
 install: $(LANGUAGES)
 	for L in $(LANGUAGES); \
-	do install -v -m 0644 $$L.mo "$(LOCALEDIR)/$$L/LC_MESSAGES/gimp20-$(PLUGIN)-plug-in.mo"; \
+	do mkdir -p "$(LOCALEDIR)/$$L/LC_MESSAGES"; \
+	install -v -m 0644 $$L.mo "$(LOCALEDIR)/$$L/LC_MESSAGES/gimp20-$(PLUGIN)-plug-in.mo"; \
 done
