--- a/setup.py
+++ b/setup.py
@@ -44,8 +44,11 @@ def build():
     _build_py = pipcl.git_info_py(".", check=0, prefix="pymupdf4llm_git_")
     ret.append((_build_py.encode(), "pymupdf4llm/_build.py"))
 
-    for p in pipcl.git_items("src"):
-        ret.append((f"src/{p}", f"pymupdf4llm/{p}"))
+    for root, _dirs, files in os.walk("src"):
+        for filename in sorted(files):
+            path = os.path.join(root, filename)
+            relative = os.path.relpath(path, "src")
+            ret.append((path, f"pymupdf4llm/{relative}"))
 
     print(f"ret:")
     for i in ret:
