Index: gtk+3.0-3.3.20/gtk/gtk.symbols
===================================================================
--- gtk+3.0-3.3.20.orig/gtk/gtk.symbols	2012-03-20 18:48:18.165061459 +0100
+++ gtk+3.0-3.3.20/gtk/gtk.symbols	2012-03-20 18:48:33.233062189 +0100
@@ -3923,6 +3923,7 @@
 gtk_window_unstick
 gtk_wrap_mode_get_type
 ubuntu_gtk_menu_shell_activate_mnemonic
+ubuntu_gtk_widget_set_has_grab
 ubuntu_menu_proxy_get_type
 ubuntu_menu_proxy_get
 ubuntu_menu_proxy_insert
Index: gtk+3.0-3.3.20/gtk/gtkwidget.c
===================================================================
--- gtk+3.0-3.3.20.orig/gtk/gtkwidget.c	2012-03-19 03:14:37.000000000 +0100
+++ gtk+3.0-3.3.20/gtk/gtkwidget.c	2012-03-20 18:48:18.261061463 +0100
@@ -14176,3 +14176,11 @@
 {
   widget->priv->style = style;
 }
+
+void
+ubuntu_gtk_widget_set_has_grab (GtkWidget *widget,
+                                gboolean   has_grab)
+{
+  _gtk_widget_set_has_grab (widget, has_grab);
+}
+ 
Index: gtk+3.0-3.3.20/gtk/gtkwidget.h
===================================================================
--- gtk+3.0-3.3.20.orig/gtk/gtkwidget.h	2012-03-18 21:58:26.000000000 +0100
+++ gtk+3.0-3.3.20/gtk/gtkwidget.h	2012-03-20 18:48:18.261061463 +0100
@@ -887,6 +887,9 @@
                                                 GdkModifierIntent  intent);
 
 
+void   ubuntu_gtk_widget_set_has_grab  (GtkWidget *widget,
+                                        gboolean   has_grab);
+
 G_END_DECLS
 
 #endif /* __GTK_WIDGET_H__ */
