--- avxutils/Makefile.am	2013-12-24 12:50:16.226794290 +0000
+++ avxutils/Makefile.am	2013-12-24 14:59:32.997468083 +0000
@@ -21,6 +21,3 @@
 
 libavxutils_la_LDFLAGS = -version-info 0:0:0 $(AM_LDFLAGS)
 
-# For some reason libtool refuses to run ldconfig
-install-exec-hook:
-	-ldconfig

--- avxsynth/Makefile.am	2013-12-24 12:50:16.214794290 +0000
+++ avxsynth/Makefile.am	2013-12-24 15:15:16.092428422 +0000
@@ -134,6 +134,4 @@
 
 libavxsynth_la_LDFLAGS = -version-info 0:0:0 -export-dynamic $(AM_LDFLAGS)
 
-# For some reason libtool refuses to run ldconfig
-install-exec-hook:
-	-ldconfig
+
