--- a/src/3rdparty/webkit/WebKit/qt/Api/qwebframe.cpp
+++ b/src/3rdparty/webkit/WebKit/qt/Api/qwebframe.cpp
@@ -317,4 +317,6 @@
     javaScriptWindowObjectCleared() signal.
 
+    If Javascript is not enabled for this page, then this method does nothing.
+
     The \a object will never be explicitly deleted by QtWebKit.
 */
@@ -339,8 +341,13 @@
     javaScriptWindowObjectCleared() signal.
 
+    If Javascript is not enabled for this page, then this method does nothing.
+
     The ownership of \a object is specified using \a own.
 */
 void QWebFrame::addToJavaScriptWindowObject(const QString &name, QObject *object, QScriptEngine::ValueOwnership ownership)
 {
+    if (!page()->settings()->testAttribute(QWebSettings::JavascriptEnabled))
+        return;
+
     JSC::JSLock lock(false);
     JSDOMWindow* window = toJSDOMWindow(d->frame);
