Index: gtk+-2.24.14/gtk/gtksettings.c
===================================================================
--- gtk+-2.24.14.orig/gtk/gtksettings.c	2013-01-05 02:33:44.407158516 +0100
+++ gtk+-2.24.14/gtk/gtksettings.c	2013-01-05 02:33:44.407158516 +0100
@@ -315,7 +315,7 @@
                                              g_param_spec_string ("gtk-fallback-icon-theme",
 								  P_("Fallback Icon Theme Name"),
 								  P_("Name of a icon theme to fall back to"),
-								  NULL,
+								  "gnome",
 								  GTK_PARAM_READWRITE),
                                              NULL);
   g_assert (result == PROP_FALLBACK_ICON_THEME);
