Index: gtk+-2.24.14/gtk/gtktreeview.c
===================================================================
--- gtk+-2.24.14.orig/gtk/gtktreeview.c	2013-01-05 02:33:51.899246761 +0100
+++ gtk+-2.24.14/gtk/gtktreeview.c	2013-01-05 02:33:51.899246761 +0100
@@ -9626,7 +9626,8 @@
 
       if (cursor_path)
 	{
-	  if (tree_view->priv->selection->type == GTK_SELECTION_MULTIPLE)
+	  if (tree_view->priv->selection->type == GTK_SELECTION_MULTIPLE ||
+	      tree_view->priv->selection->type == GTK_SELECTION_SINGLE)
 	    gtk_tree_view_real_set_cursor (tree_view, cursor_path, FALSE, FALSE);
 	  else
 	    gtk_tree_view_real_set_cursor (tree_view, cursor_path, TRUE, FALSE);
