diff --git a/mkspecs/common/linux.conf b/mkspecs/common/linux.conf
index 1ae5608..9c603a6 100644
--- a/mkspecs/common/linux.conf
+++ b/mkspecs/common/linux.conf
@@ -7,12 +7,12 @@ QMAKE_CXXFLAGS_THREAD	+= $$QMAKE_CFLAGS_THREAD
 
 QMAKE_INCDIR          =
 QMAKE_LIBDIR          =
-QMAKE_INCDIR_X11      = /usr/X11R6/include
-QMAKE_LIBDIR_X11      = /usr/X11R6/lib
+QMAKE_INCDIR_X11      = /usr/include
+QMAKE_LIBDIR_X11      = /usr/lib
 QMAKE_INCDIR_QT       = $$[QT_INSTALL_HEADERS]
 QMAKE_LIBDIR_QT       = $$[QT_INSTALL_LIBS]
-QMAKE_INCDIR_OPENGL   = /usr/X11R6/include
-QMAKE_LIBDIR_OPENGL   = /usr/X11R6/lib
+QMAKE_INCDIR_OPENGL   = /usr/include
+QMAKE_LIBDIR_OPENGL   = /usr/lib
 QMAKE_INCDIR_OPENGL_ES1 = $$QMAKE_INCDIR_OPENGL
 QMAKE_LIBDIR_OPENGL_ES1 = $$QMAKE_LIBDIR_OPENGL
 QMAKE_INCDIR_OPENGL_ES1CL = $$QMAKE_INCDIR_OPENGL
diff --git a/mkspecs/linux-cxx/qmake.conf b/mkspecs/linux-cxx/qmake.conf
index 633b738..a37c05b 100644
--- a/mkspecs/linux-cxx/qmake.conf
+++ b/mkspecs/linux-cxx/qmake.conf
@@ -37,12 +37,12 @@ QMAKE_CXXFLAGS_YACC	= $$QMAKE_CFLAGS_YACC
 
 QMAKE_INCDIR		=
 QMAKE_LIBDIR		=
-QMAKE_INCDIR_X11	= /usr/X11R6/include
-QMAKE_LIBDIR_X11	= /usr/X11R6/lib
+QMAKE_INCDIR_X11	= /usr/include
+QMAKE_LIBDIR_X11	= /usr/lib
 QMAKE_INCDIR_QT		= $$[QT_INSTALL_HEADERS]
 QMAKE_LIBDIR_QT		= $$[QT_INSTALL_LIBS]
-QMAKE_INCDIR_OPENGL	= /usr/X11R6/include
-QMAKE_LIBDIR_OPENGL	= /usr/X11R6/lib
+QMAKE_INCDIR_OPENGL	= /usr/include
+QMAKE_LIBDIR_OPENGL	= /usr/lib
 
 QMAKE_LINK		= cxx
 QMAKE_LINK_SHLIB	= cxx
diff --git a/mkspecs/linux-ecc-64/qmake.conf b/mkspecs/linux-ecc-64/qmake.conf
index 359d44b..2d3f9a5 100644
--- a/mkspecs/linux-ecc-64/qmake.conf
+++ b/mkspecs/linux-ecc-64/qmake.conf
@@ -39,12 +39,12 @@ QMAKE_CXXFLAGS_THREAD	= $$QMAKE_CFLAGS_THREAD
 
 QMAKE_INCDIR		=
 QMAKE_LIBDIR		=
-QMAKE_INCDIR_X11	= /usr/X11R6/include
-QMAKE_LIBDIR_X11	= /usr/X11R6/lib
+QMAKE_INCDIR_X11	= /usr/include
+QMAKE_LIBDIR_X11	= /usr/lib
 QMAKE_INCDIR_QT		= $$[QT_INSTALL_HEADERS]
 QMAKE_LIBDIR_QT		= $$[QT_INSTALL_LIBS]
-QMAKE_INCDIR_OPENGL	= /usr/X11R6/include
-QMAKE_LIBDIR_OPENGL	= /usr/X11R6/lib
+QMAKE_INCDIR_OPENGL	= /usr/include
+QMAKE_LIBDIR_OPENGL	= /usr/lib
 
 QMAKE_LINK		= ecpc
 QMAKE_LINK_SHLIB	= ecpc
diff --git a/mkspecs/linux-g++-64/qmake.conf b/mkspecs/linux-g++-64/qmake.conf
index 4f8794f..cf7be62 100644
--- a/mkspecs/linux-g++-64/qmake.conf
+++ b/mkspecs/linux-g++-64/qmake.conf
@@ -17,7 +17,7 @@ QMAKE_LFLAGS		= -m64
 include(../common/g++.conf)
 include(../common/linux.conf)
 
-QMAKE_LIBDIR_X11      = /usr/X11R6/lib64
-QMAKE_LIBDIR_OPENGL   = /usr/X11R6/lib64
+QMAKE_LIBDIR_X11      = /usr/lib64
+QMAKE_LIBDIR_OPENGL   = /usr/lib64
 
 load(qt_config)
diff --git a/mkspecs/linux-icc-64/qmake.conf b/mkspecs/linux-icc-64/qmake.conf
index 7962026..9b4bade 100644
--- a/mkspecs/linux-icc-64/qmake.conf
+++ b/mkspecs/linux-icc-64/qmake.conf
@@ -12,5 +12,5 @@ include(../linux-icc/qmake.conf)
 
 # Change the all LIBDIR variables to use lib64 instead of lib
 
-QMAKE_LIBDIR_X11	= /usr/X11R6/lib64
-QMAKE_LIBDIR_OPENGL	= /usr/X11R6/lib64
+QMAKE_LIBDIR_X11	= /usr/lib64
+QMAKE_LIBDIR_OPENGL	= /usr/lib64
diff --git a/mkspecs/linux-icc/qmake.conf b/mkspecs/linux-icc/qmake.conf
index 965de0c..b778a89 100644
--- a/mkspecs/linux-icc/qmake.conf
+++ b/mkspecs/linux-icc/qmake.conf
@@ -45,12 +45,12 @@ QMAKE_CXXFLAGS_THREAD	= $$QMAKE_CFLAGS_THREAD
 
 QMAKE_INCDIR		=
 QMAKE_LIBDIR		=
-QMAKE_INCDIR_X11	= /usr/X11R6/include
-QMAKE_LIBDIR_X11	= /usr/X11R6/lib
+QMAKE_INCDIR_X11	= /usr/include
+QMAKE_LIBDIR_X11	= /usr/lib
 QMAKE_INCDIR_QT		= $$[QT_INSTALL_HEADERS]
 QMAKE_LIBDIR_QT		= $$[QT_INSTALL_LIBS]
-QMAKE_INCDIR_OPENGL	= /usr/X11R6/include
-QMAKE_LIBDIR_OPENGL	= /usr/X11R6/lib
+QMAKE_INCDIR_OPENGL	= /usr/include
+QMAKE_LIBDIR_OPENGL	= /usr/lib
 
 QMAKE_LINK		= icpc
 QMAKE_LINK_SHLIB	= icpc
diff --git a/mkspecs/linux-kcc/qmake.conf b/mkspecs/linux-kcc/qmake.conf
index 009c486..0e3d301 100644
--- a/mkspecs/linux-kcc/qmake.conf
+++ b/mkspecs/linux-kcc/qmake.conf
@@ -48,12 +48,12 @@ QMAKE_CXXFLAGS_THREAD	= $$QMAKE_CFLAGS_THREAD
 
 QMAKE_INCDIR		=
 QMAKE_LIBDIR		=
-QMAKE_INCDIR_X11	= /usr/X11R6/include
-QMAKE_LIBDIR_X11	= /usr/X11R6/lib
+QMAKE_INCDIR_X11	= /usr/include
+QMAKE_LIBDIR_X11	= /usr/lib
 QMAKE_INCDIR_QT		= $$[QT_INSTALL_HEADERS]
 QMAKE_LIBDIR_QT		= $$[QT_INSTALL_LIBS]
-QMAKE_INCDIR_OPENGL	= /usr/X11R6/include
-QMAKE_LIBDIR_OPENGL	= /usr/X11R6/lib
+QMAKE_INCDIR_OPENGL	= /usr/include
+QMAKE_LIBDIR_OPENGL	= /usr/lib
 
 QMAKE_LINK		= KCC
 QMAKE_LINK_SHLIB	= KCC
diff --git a/mkspecs/linux-pgcc/qmake.conf b/mkspecs/linux-pgcc/qmake.conf
index 756f1af..2c40635 100644
--- a/mkspecs/linux-pgcc/qmake.conf
+++ b/mkspecs/linux-pgcc/qmake.conf
@@ -39,12 +39,12 @@ QMAKE_CXXFLAGS_THREAD	= $$QMAKE_CFLAGS_THREAD
 
 QMAKE_INCDIR		=
 QMAKE_LIBDIR		=
-QMAKE_INCDIR_X11	= /usr/X11R6/include
-QMAKE_LIBDIR_X11	= /usr/X11R6/lib
+QMAKE_INCDIR_X11	= /usr/include
+QMAKE_LIBDIR_X11	= /usr/lib
 QMAKE_INCDIR_QT		= $$[QT_INSTALL_HEADERS]
 QMAKE_LIBDIR_QT		= $$[QT_INSTALL_LIBS]
-QMAKE_INCDIR_OPENGL	= /usr/X11R6/include
-QMAKE_LIBDIR_OPENGL	= /usr/X11R6/lib
+QMAKE_INCDIR_OPENGL	= /usr/include
+QMAKE_LIBDIR_OPENGL	= /usr/lib
 
 QMAKE_LINK		= pgCC
 QMAKE_LINK_SHLIB	= pgCC
