--- release/freedesktop/blender.desktop	2012-04-18 02:08:47.000000000 -0700
+++ release/freedesktop/blender.desktop	2012-04-18 02:17:13.000000000 -0700
@@ -1,5 +1,5 @@
 [Desktop Entry]
-Name=Blender
+Name=Blender-2.6
 GenericName=3D modeller
 GenericName[es]=modelador 3D
 GenericName[de]=3D-Modellierer
@@ -8,7 +8,7 @@
 Comment=3D modeling, animation, rendering and post-production
 Comment[es]=modelado 3D, animaciГіn, renderizado y post-producciГіn
 Comment[de]=3D-Modellierung, Animation, Rendering und Nachbearbeitung
-Exec=blender
+Exec=blender-2.6 %f
 Icon=blender
 Terminal=false
 Type=Application
