diff -Naur compiz-0.8.8/kde/window-decorator-kde4/window.cpp compiz-0.8.8-gcc-4.7/kde/window-decorator-kde4/window.cpp
--- compiz-0.8.8/kde/window-decorator-kde4/window.cpp	2010-05-21 11:18:14.000000000 +0000
+++ compiz-0.8.8-gcc-4.7/kde/window-decorator-kde4/window.cpp	2012-09-12 08:54:35.000000000 +0000
@@ -59,6 +59,9 @@
 
 #include "paintredirector.h"
 
+/* gcc-4.7 hack */
+#include "unistd.h"
+
 KWD::Window::Window (WId  parentId,
 		     WId  clientId,
 		     WId  frame,
