diff -Nur gtk+3.0-3.6.0.old/gtk/a11y/gail.c gtk+3.0-3.6.0/gtk/a11y/gail.c
--- gtk+3.0-3.6.0.old/gtk/a11y/gail.c	2012-10-09 11:28:38.129582008 +0200
+++ gtk+3.0-3.6.0/gtk/a11y/gail.c	2012-10-09 11:30:27.485587299 +0200
@@ -76,7 +76,6 @@
 static guint focus_notify_handler = 0;    
 static guint focus_tracker_id = 0;
 static GQuark quark_focus_object = 0;
-static int initialized = FALSE;
 
 static AtkObject*
 gail_get_accessible_for_widget (GtkWidget *widget,
@@ -796,24 +795,9 @@
 }
 
 void
-_gtk_accessibility_shutdown (void)
-{
-  if (!initialized)
-    return;
-
-  initialized = FALSE;
-
-  g_clear_object (&atk_misc_instance);
-
-#ifdef GDK_WINDOWING_X11
-  atk_bridge_adaptor_cleanup ();
-#endif
-  _gail_util_uninstall ();
-}
-
-void
 _gtk_accessibility_init (void)
 {
+  static int initialized = FALSE;
 
   if (initialized)
     return;
diff -Nur gtk+3.0-3.6.0.old/gtk/a11y/gailutil.c gtk+3.0-3.6.0/gtk/a11y/gailutil.c
--- gtk+3.0-3.6.0.old/gtk/a11y/gailutil.c	2012-10-09 11:28:38.117582007 +0200
+++ gtk+3.0-3.6.0/gtk/a11y/gailutil.c	2012-10-09 11:30:27.485587299 +0200
@@ -206,17 +206,6 @@
                     (GCallback) window_removed, NULL);
 }
 
-static void
-undo_window_event_initialization (void)
-{
-  AtkObject *root;
-
-  root = atk_get_root ();
-
-  g_signal_handlers_disconnect_by_func (root, (GCallback) window_added, NULL);
-  g_signal_handlers_disconnect_by_func (root, (GCallback) window_removed, NULL);
-}
-
 static AtkKeyEventStruct *
 atk_key_event_from_gdk_event_key (GdkEventKey *key)
 {
@@ -353,12 +342,6 @@
 }
 
 void
-_gail_util_uninstall (void)
-{
-  undo_window_event_initialization ();
-}
-
-void
 _gail_util_install (void)
 {
   AtkUtilClass *atk_class = ATK_UTIL_CLASS (g_type_class_ref (ATK_TYPE_UTIL));
diff -Nur gtk+3.0-3.6.0.old/gtk/a11y/gailutil.h gtk+3.0-3.6.0/gtk/a11y/gailutil.h
--- gtk+3.0-3.6.0.old/gtk/a11y/gailutil.h	2012-10-09 11:28:38.121582007 +0200
+++ gtk+3.0-3.6.0/gtk/a11y/gailutil.h	2012-10-09 11:30:27.485587299 +0200
@@ -23,7 +23,6 @@
 G_BEGIN_DECLS
 
 void _gail_util_install   (void);
-void _gail_util_uninstall (void);
 
 gboolean  _gail_util_key_snooper (GtkWidget   *the_widget,
                                   GdkEventKey *event);
diff -Nur gtk+3.0-3.6.0.old/gtk/gtkapplication.c gtk+3.0-3.6.0/gtk/gtkapplication.c
--- gtk+3.0-3.6.0.old/gtk/gtkapplication.c	2012-10-09 11:28:38.261582015 +0200
+++ gtk+3.0-3.6.0/gtk/gtkapplication.c	2012-10-09 11:30:27.493587302 +0200
@@ -47,8 +47,6 @@
 #include <gdk/x11/gdkx.h>
 #endif
 
-extern void _gtk_accessibility_shutdown (void);
-
 /**
  * SECTION:gtkapplication
  * @title: GtkApplication
@@ -425,16 +423,12 @@
   gtk_application_shutdown_quartz (GTK_APPLICATION (application));
 #endif
 
-  /* Keep this section in sync with gtk_main() */
-
   /* Try storing all clipboard data we have */
   _gtk_clipboard_store_all ();
 
   /* Synchronize the recent manager singleton */
   _gtk_recent_manager_sync ();
 
-  _gtk_accessibility_shutdown ();
-
   G_APPLICATION_CLASS (gtk_application_parent_class)
     ->shutdown (application);
 }
diff -Nur gtk+3.0-3.6.0.old/gtk/gtkmain.c gtk+3.0-3.6.0/gtk/gtkmain.c
--- gtk+3.0-3.6.0.old/gtk/gtkmain.c	2012-10-09 11:28:38.217582012 +0200
+++ gtk+3.0-3.6.0/gtk/gtkmain.c	2012-10-09 11:30:27.497587303 +0200
@@ -1173,15 +1172,11 @@
 
   if (gtk_main_loop_level == 0)
     {
-      /* Keep this section in sync with gtk_application_shutdown() */
-
       /* Try storing all clipboard data we have */
       _gtk_clipboard_store_all ();
 
       /* Synchronize the recent manager singleton */
       _gtk_recent_manager_sync ();
-
-      _gtk_accessibility_shutdown ();
     }
 }
 
