--- cuda-z.pro.orig	2014-11-28 19:10:27.000000000 +0300
+++ cuda-z.pro	2014-11-28 19:20:40.863774724 +0300
@@ -88,8 +88,7 @@
 	QMAKE_RC = rc $$join(RC_INCLUDEPATH, " -I ", "-I ")
 }
 
-unix:LIBS += -lcudart_static
-linux:!static:LIBS += -ldl -lm -lrt
+unix:LIBS += -L/opt/cuda/lib64 -lcudart -ldl
 win32:LIBS += \
 	$$quote($$(CUDA_LIB_PATH)\\cuda.lib) \
 	$$quote($$(CUDA_LIB_PATH)\\cudart_static.lib) \
