--- makefile~	2009-02-23 19:37:52.000000000 +0100
+++ makefile	2009-04-04 23:23:33.000000000 +0200
@@ -18,7 +18,7 @@
 CXXFLAGS += `sdl-config --cflags` -DVERSION=$(VERSION) -DRELEASE=$(RELEASE) -DUSEPAK=$(USEPAK)
 CXXFLAGS += -DPAKNAME=\"$(PAKNAME)\" -DPAKLOCATION=\"$(DATADIR)\" -DUNIX -DGAMEPLAYMANUAL=\"$(DOCDIR)index.html\" -Wall
 CXXFLAGS += -DLOCALEDIR=\"$(LOCALEDIR)\"
-CXXFLAGS += $(CFLAGS) -Werror
+CXXFLAGS += $(CFLAGS)
 LIBS = `sdl-config --libs` -lSDL_mixer -lSDL_image -lSDL_ttf -lz
 
 OBJS += CAudio.o
