*** src/pngio.c	2010-09-18 19:20:42.000000000 +0200
--- src/pngio.c.new	2011-12-05 12:22:06.000000000 +0100
***************
*** 106,111 ****
--- 106,112 ----
  /* --------------------------------------------*/
  
  #include "png.h"
+ #include "zlib.h"
  
  /* ----------------Set defaults for read/write options ----------------- */
      /* strip 16 bpp --> 8 bpp on reading png; default is for stripping */
