--- src/libOpenImageIO/CMakeLists.txt.orig	2014-11-12 18:06:25.000000000 +0300
+++ src/libOpenImageIO/CMakeLists.txt	2014-11-12 18:08:17.302269164 +0300
@@ -2,9 +2,9 @@
 
 if (NOT USE_EXTERNAL_PUGIXML)
     list (APPEND libOpenImageIO_hdrs
-          ../include/OpenImageIO/pugiconfig.hpp
-          ../include/OpenImageIO/pugixml.hpp
-          ../include/OpenImageIO/pugixml.cpp
+          /usr/include/pugiconfig.hpp
+          /usr/include/pugixml.hpp
+          /usr/include/pugixml.cpp
     )
 endif()
 
--- src/libOpenImageIO/formatspec.cpp.org	2014-11-12 18:11:27.000000000 +0300
+++ src/libOpenImageIO/formatspec.cpp	2014-11-12 18:12:32.823290614 +0300
@@ -43,7 +43,7 @@
 #include "OpenImageIO/fmath.h"
 #include "OpenImageIO/imageio.h"
 #include "imageio_pvt.h"
-#include "OpenImageIO/pugixml.hpp"
+#include "/usr/include/pugixml.hpp"
 
 
 OIIO_NAMESPACE_ENTER

--- src/libOpenImageIO/xmp.cpp.org	2014-11-12 18:14:50.000000000 +0300
+++ src/libOpenImageIO/xmp.cpp	2014-11-12 18:15:28.503305361 +0300
@@ -38,7 +38,7 @@
 #include "OpenImageIO/strutil.h"
 #include "OpenImageIO/fmath.h"
 #include "OpenImageIO/imageio.h"
-#include "OpenImageIO/pugixml.hpp"
+#include "/usr/include/pugixml.hpp"
 
 #define DEBUG_XMP_READ  0
 #define DEBUG_XMP_WRITE 0
 