--- a/src/napi4.c
+++ b/src/napi4.c
@@ -70,6 +70,19 @@
     assert(pRes->iNXID == NXSIGNATURE);
     return pRes;
   }
+
+static int32 NXIgetname(int32 vgroup, char *name)
+{
+  size_t size = sizeof(NXname);
+  return Vgetname(vgroup, name, &size);
+}
+
+static int32 NXIgetclass(int32 vgroup, char *name)
+{
+  size_t size = sizeof(NXname);
+  return Vgetclass(vgroup, name, &size);
+}
+
   /*----------------------------------------------------------------------*/
 static int findNapiClass(pNexusFile pFile, int groupRef, NXname nxclass)
 {
@@ -77,7 +90,7 @@
   int32 tags[2], attID, linkID, groupID;
 
   groupID = Vattach(pFile->iVID,groupRef,"r");
-  Vgetclass(groupID, classText);
+  NXIgetclass(groupID, classText);
   if(strcmp(classText,"NAPIlink") != 0)
   {
     /* normal group */
@@ -93,7 +106,7 @@
     {
       Vgetattr(groupID,attID, tags);
       linkID = Vattach(pFile->iVID,tags[1],"r");
-      Vgetclass(linkID, linkClass);
+      NXIgetclass(linkID, linkClass);
       Vdetach(groupID);
       Vdetach(linkID);
       strcpy(nxclass,linkClass);
@@ -135,7 +148,7 @@
       /* loop and check */
       for (i = 0; i < iN; i++) {
         iNew = Vattach (pFile->iVID, pArray[i], "r");
-        Vgetname (iNew, pText);
+        NXIgetname(iNew, pText);
         Vdetach(iNew);
         if (strcmp (pText, name) == 0) {
           pArray[i] = findNapiClass(pFile,pArray[i],pText);
@@ -156,7 +169,7 @@
         Vgettagref (pFile->iCurrentVG, i, &iTag, &iRef);
         if (iTag == DFTAG_VG) {
           iNew = Vattach (pFile->iVID, iRef, "r");
-          Vgetname (iNew, pText);
+          NXIgetname(iNew, pText);
           Vdetach(iNew);
           if (strcmp (pText, name) == 0) {
 	    iRef = findNapiClass(pFile,iRef, pText);
@@ -351,7 +364,7 @@
       groupID = Vattach(pFile->iVID,pFile->iStack[i].iVref, "r");
       if (groupID != -1)
       {
-          if (Vgetname(groupID, pText) != -1) {
+          if (NXIgetname(groupID, pText) != -1) {
               strncat(buffer,pText,bufLen-strlen(buffer));
           } else {
               NXReportError( "ERROR: NXIbuildPath cannot get vgroup name");
@@ -1527,7 +1540,7 @@
         NXReportError( "ERROR: HDF cannot attach to Vgroup");
         return NX_ERROR;
       }
-      Vgetname (iTemp, name);
+      NXIgetname(iTemp, name);
       Vdetach (iTemp);
       findNapiClass(pFile, pFile->iStack[pFile->iStackPtr].iRefDir[iCurDir], nxclass);
       *datatype = DFTAG_VG;
@@ -1541,7 +1554,7 @@
           NXReportError( "ERROR: HDF cannot attach to Vgroup");
           return NX_ERROR;
         }
-        Vgetname (iTemp, name);
+        NXIgetname(iTemp, name);
         Vdetach(iTemp);
 	findNapiClass(pFile, pFile->iStack[pFile->iStackPtr].iRefDir[iCurDir], nxclass);
         *datatype = DFTAG_VG;
@@ -1908,8 +1921,8 @@
     }
     else {
       *iN = Vntagrefs (pFile->iCurrentVG);
-      Vgetname (pFile->iCurrentVG, pName);
-      Vgetclass (pFile->iCurrentVG, pClass);
+      NXIgetname(pFile->iCurrentVG, pName);
+      NXIgetclass(pFile->iCurrentVG, pClass);
     }
     return NX_OK;
   }
