Description: fix numpy install dirs as set in Debian
Author: Mathieu Malaterre <malat@debian.org>
Forwarded: not-needed

Index: openvdb/Makefile
===================================================================
--- openvdb.orig/Makefile	2013-12-29 12:31:47.686961360 +0100
+++ openvdb/Makefile	2013-12-29 12:31:51.382961441 +0100
@@ -465,7 +465,7 @@
 #
 PYCXXFLAGS := -fPIC -I python -I $(PYTHON_INCL_DIR) -I $(PYCONFIG_INCL_DIR)
 ifneq ($(strip $(NUMPY_INCL_DIR)),)
-PYCXXFLAGS += -I $(NUMPY_INCL_DIR) -DPY_OPENVDB_USE_NUMPY
+PYCXXFLAGS += -I $(NUMPY_INCL_DIR) -I $(NUMPY_INCL_DIR2) -DPY_OPENVDB_USE_NUMPY
 endif
 ifneq (no,$(strip $(PYTHON_WRAP_ALL_GRID_TYPES)))
 PYCXXFLAGS += -DPY_OPENVDB_WRAP_ALL_GRID_TYPES
