--- openvdb/python/pyGrid.h
+++ openvdb/python/pyGrid.h
@@ -41,6 +41,7 @@
 #ifdef PY_OPENVDB_USE_NUMPY
 #define PY_ARRAY_UNIQUE_SYMBOL PY_OPENVDB_ARRAY_API
 #define NO_IMPORT_ARRAY // NumPy gets initialized during module initialization
+#define NPY_NO_DEPRECATED_API NPY_1_7_API_VERSION
 #include <arrayobject.h> // for PyArrayObject
 #include "openvdb/tools/MeshToVolume.h"
 #include "openvdb/tools/VolumeToMesh.h" // for tools::volumeToMesh()
