--- openvdb/python/pyOpenVDBModule.cc
+++ openvdb/python/pyOpenVDBModule.cc
@@ -35,6 +35,7 @@
 #include <boost/python/exception_translator.hpp>
 #ifdef PY_OPENVDB_USE_NUMPY
 #define PY_ARRAY_UNIQUE_SYMBOL PY_OPENVDB_ARRAY_API
+#define NPY_NO_DEPRECATED_API NPY_1_7_API_VERSION
 #include <arrayobject.h> // for import_array()
 #endif
 #include "openvdb/openvdb.h"
@@ -596,7 +596,7 @@
 
 #ifdef PY_OPENVDB_USE_NUMPY
     // Initialize NumPy.
-    import_array();
+    if (_import_array()) { };
 #endif
 
     using namespace openvdb::OPENVDB_VERSION_NAME;
