--- include/host_config.h.orig
+++ include/host_config.h
@@ -77,11 +77,11 @@
 
 #if defined(__GNUC__)
 
-#if __GNUC__ > 4 || (__GNUC__ == 4 && __GNUC_MINOR__ > 8)
+#if __GNUC__ > 4 || (__GNUC__ == 4 && __GNUC_MINOR__ > 9)
 
-#error -- unsupported GNU version! gcc 4.9 and up are not supported!
+#error -- unsupported GNU version! gcc up 4.9 are not supported!
 
-#endif /* __GNUC__> 4 || (__GNUC__ == 4 && __GNUC_MINOR__ > 8) */
+#endif /* __GNUC__> 4 || (__GNUC__ == 4 && __GNUC_MINOR__ > 9) */
 
 #endif /* __GNUC__ */
 
