diff -Naur GrabVK/GrabVK.desktop GrabVK2/GrabVK.desktop
--- GrabVK/grabvk.desktop	1970-01-01 00:00:00.000000000 +0000
+++ GrabVK2/grabvk.desktop	2010-12-15 08:21:37.000000000 +0000
@@ -0,0 +1,13 @@
+[Desktop Entry]
+X-Desktop-File-Install-Version=0.0.4
+Name=GrabVK
+Comment=music downloader from vkontakte.ru
+GenericName=music downloader from vkontakte.ru
+Exec=GrabVK
+Icon=grabVK
+Terminal=false
+Type=Application
+Categories=AudioVideo;Player;Audio;Qt;
+MimeType=
+X-KDE-StartupNotify=false
+
