Index: chrome/browser/autofill/autofill_country.cc
diff --git a/chrome/browser/autofill/autofill_country.cc b/chrome/browser/autofill/autofill_country.cc
index a0f1bc8fcfe4a8f90f9845a101dc90959176600a..f39c6511f0e39f55bf972015c35e6919f8cd7266 100644
--- a/chrome/browser/autofill/autofill_country.cc
+++ b/chrome/browser/autofill/autofill_country.cc
@@ -13,12 +13,12 @@
 #include "base/utf_string_conversions.h"
 #include "chrome/browser/browser_process.h"
 #include "grit/generated_resources.h"
-#include "third_party/icu/public/common/unicode/locid.h"
-#include "third_party/icu/public/common/unicode/uloc.h"
-#include "third_party/icu/public/i18n/unicode/coll.h"
-#include "third_party/icu/public/i18n/unicode/ucol.h"
 #include "ui/base/l10n/l10n_util_collator.h"
 #include "ui/base/l10n/l10n_util.h"
+#include "unicode/coll.h"
+#include "unicode/locid.h"
+#include "unicode/ucol.h"
+#include "unicode/uloc.h"
 
 namespace {
 
