diff -Naur /var/tmp/portage/www-client/chromium-16.0.882.0/work/chromium-16.0.882.0/chrome/chrome_browser.gypi chromium-16.0.882.0/chrome/chrome_browser.gypi
--- /var/tmp/portage/www-client/chromium-16.0.882.0/work/chromium-16.0.882.0/chrome/chrome_browser.gypi	2011-09-16 08:33:24.000000000 +0000
+++ chromium-16.0.882.0/chrome/chrome_browser.gypi	2011-09-16 19:47:11.000000000 +0000
@@ -63,6 +63,7 @@
       ],
       'include_dirs': [
         '..',
+        '/usr/include/dbus-1.0/',
         '<(INTERMEDIATE_DIR)',
       ],
       'defines': [