--- fixed/chromium-15.0.846.0//third_party/webrtc/modules/audio_device/main/source/audio_device.gyp	2011-08-07 08:36:26.000000000 +0000
+++ chromium-15.0.846.0//third_party/webrtc/modules/audio_device/main/source/audio_device.gyp	2011-08-07 17:31:48.000000000 +0000
@@ -130,26 +130,6 @@
               '-lasound',
             ],
           },
-          'conditions': [
-            ['include_pulse_audio==1', {
-              'defines': [
-                'LINUX_PULSE',
-              ],
-              'sources': [
-                'Linux/audio_device_linux_pulse.cc',
-                'Linux/audio_device_linux_pulse.h',
-                'Linux/audio_mixer_manager_linux_pulse.cc',
-                'Linux/audio_mixer_manager_linux_pulse.h',
-                'Linux/pulseaudiosymboltable.cc',
-                'Linux/pulseaudiosymboltable.h',
-              ],
-              'link_settings': {
-                'libraries': [
-                  '-lpulse',
-                ],
-              },
-            }],
-          ],
         }],
         ['OS=="mac"', {
           'include_dirs': [
diff -Naur fixed/chromium-15.0.846.0//third_party/webrtc/video_engine/main/test/AutoTest/vie_auto_test.gypi chromium-15.0.846.0//third_party/webrtc/video_engine/main/test/AutoTest/vie_auto_test.gypi
--- fixed/chromium-15.0.846.0//third_party/webrtc/video_engine/main/test/AutoTest/vie_auto_test.gypi	2011-08-07 08:36:22.000000000 +0000
+++ chromium-15.0.846.0//third_party/webrtc/video_engine/main/test/AutoTest/vie_auto_test.gypi	2011-08-07 17:23:46.000000000 +0000
@@ -110,7 +110,6 @@
             '-lXext',
             '-lX11',
             '-lasound',
-            '-lpulse',
 
 
           ],
