--- chromium-14.0.835.18//third_party/webrtc/modules/audio_device/main/source/Linux/audio_device_linux_pulse.cc	2011-08-06 17:01:03.000000000 +0000
+++ fixed/chromium-14.0.835.18/work/chromium-14.0.835.18//third_party/webrtc/modules/audio_device/main/source/Linux/audio_device_linux_pulse.cc	2011-08-06 17:07:51.000000000 +0000
@@ -11,7 +11,6 @@
 #include <cassert>
 
 #include "audio_device_utility.h"
-#include "audio_device_linux_pulse.h"
 #include "audio_device_config.h"
 
 #include "event_wrapper.h"
diff -Naur chromium-14.0.835.18//third_party/webrtc/modules/audio_device/main/source/Linux/audio_device_linux_pulse.h fixed/chromium-14.0.835.18/work/chromium-14.0.835.18//third_party/webrtc/modules/audio_device/main/source/Linux/audio_device_linux_pulse.h
--- chromium-14.0.835.18//third_party/webrtc/modules/audio_device/main/source/Linux/audio_device_linux_pulse.h	2011-08-06 17:01:03.000000000 +0000
+++ fixed/chromium-14.0.835.18/work/chromium-14.0.835.18//third_party/webrtc/modules/audio_device/main/source/Linux/audio_device_linux_pulse.h	2011-08-06 17:06:04.000000000 +0000
@@ -12,10 +12,8 @@
 #define WEBRTC_AUDIO_DEVICE_AUDIO_DEVICE_LINUX_PULSE_H
 
 #include "audio_device_generic.h"
-#include "audio_mixer_manager_linux_pulse.h"
 #include "critical_section_wrapper.h"
 
-#include <pulse/pulseaudio.h>
 
 // Set this define to make the code behave like in GTalk/libjingle
 //#define WEBRTC_PA_GTALK
diff -Naur chromium-14.0.835.18//third_party/webrtc/modules/audio_device/main/source/Linux/audio_mixer_manager_linux_pulse.cc fixed/chromium-14.0.835.18/work/chromium-14.0.835.18//third_party/webrtc/modules/audio_device/main/source/Linux/audio_mixer_manager_linux_pulse.cc
--- chromium-14.0.835.18//third_party/webrtc/modules/audio_device/main/source/Linux/audio_mixer_manager_linux_pulse.cc	2011-08-06 17:01:03.000000000 +0000
+++ fixed/chromium-14.0.835.18/work/chromium-14.0.835.18//third_party/webrtc/modules/audio_device/main/source/Linux/audio_mixer_manager_linux_pulse.cc	2011-08-06 17:08:16.000000000 +0000
@@ -10,7 +10,6 @@
 
 #include <cassert>
 
-#include "audio_mixer_manager_linux_pulse.h"
 #include "trace.h"
 
 extern webrtc_adm_linux_pulse::PulseAudioSymbolTable PaSymbolTable;
diff -Naur chromium-14.0.835.18//third_party/webrtc/modules/audio_device/main/source/Linux/audio_mixer_manager_linux_pulse.h fixed/chromium-14.0.835.18/work/chromium-14.0.835.18//third_party/webrtc/modules/audio_device/main/source/Linux/audio_mixer_manager_linux_pulse.h
--- chromium-14.0.835.18//third_party/webrtc/modules/audio_device/main/source/Linux/audio_mixer_manager_linux_pulse.h	2011-08-06 17:01:03.000000000 +0000
+++ fixed/chromium-14.0.835.18/work/chromium-14.0.835.18//third_party/webrtc/modules/audio_device/main/source/Linux/audio_mixer_manager_linux_pulse.h	2011-08-06 17:02:09.000000000 +0000
@@ -17,7 +17,6 @@
 #include "pulseaudiosymboltable.h"
 
 #include <stdint.h>
-#include <pulse/pulseaudio.h>
 
 #ifndef UINT32_MAX
 #define UINT32_MAX  ((uint32_t)-1)
