--- a/mm/slab.h
+++ b/mm/slab.h
@@ -58,8 +57,0 @@
-#ifdef CONFIG_64BIT
-					/*
-					 * Some optimizations use free bits in 'counters' field
-					 * to save memory. In case ->stride field is not available,
-					 * such optimizations are disabled.
-					 */
-					unsigned short stride;
-#endif
@@ -292 +283,0 @@
-int sysfs_slab_alias(struct kmem_cache *s, const char *name);
@@ -296,2 +286,0 @@
-static inline int sysfs_slab_alias(struct kmem_cache *s, const char *name)
-							{ return 0; }
@@ -413,0 +403,5 @@
+struct kmem_cache *find_mergeable(unsigned size, unsigned align,
+		slab_flags_t flags, const char *name, void (*ctor)(void *));
+struct kmem_cache *
+__kmem_cache_alias(const char *name, unsigned int size, unsigned int align,
+		   slab_flags_t flags, void (*ctor)(void *));
@@ -511,18 +504,0 @@
-/*
- * slub is about to manipulate internal object metadata.  This memory lies
- * outside the range of the allocated object, so accessing it would normally
- * be reported by kasan as a bounds error.  metadata_access_enable() is used
- * to tell kasan that these accesses are OK.
- */
-static inline void metadata_access_enable(void)
-{
-	kasan_disable_current();
-	kmsan_disable_current();
-}
-
-static inline void metadata_access_disable(void)
-{
-	kmsan_enable_current();
-	kasan_enable_current();
-}
-
@@ -536,20 +512,4 @@
- * Returns the address of the object extension vector associated with the slab,
- * or zero if no such vector has been associated yet.
- * Do not dereference the return value directly; use get/put_slab_obj_exts()
- * pair and slab_obj_ext() to access individual elements.
- *
- * Example usage:
- *
- * obj_exts = slab_obj_exts(slab);
- * if (obj_exts) {
- *         get_slab_obj_exts(obj_exts);
- *         obj_ext = slab_obj_ext(slab, obj_exts, obj_to_index(s, slab, obj));
- *         // do something with obj_ext
- *         put_slab_obj_exts(obj_exts);
- * }
- *
- * Note that the get/put semantics does not involve reference counting.
- * Instead, it updates kasan/kmsan depth so that accesses to slabobj_ext
- * won't be reported as access violations.
- */
-static inline unsigned long slab_obj_exts(struct slab *slab)
+ * Returns a pointer to the object extension vector associated with the slab,
+ * or NULL if no such vector has been associated yet.
+ */
+static inline struct slabobj_ext *slab_obj_exts(struct slab *slab)
@@ -568,56 +528 @@
-
-	return obj_exts & ~OBJEXTS_FLAGS_MASK;
-}
-
-static inline void get_slab_obj_exts(unsigned long obj_exts)
-{
-	VM_WARN_ON_ONCE(!obj_exts);
-	metadata_access_enable();
-}
-
-static inline void put_slab_obj_exts(unsigned long obj_exts)
-{
-	metadata_access_disable();
-}
-
-#ifdef CONFIG_64BIT
-static inline void slab_set_stride(struct slab *slab, unsigned short stride)
-{
-	slab->stride = stride;
-}
-static inline unsigned short slab_get_stride(struct slab *slab)
-{
-	return slab->stride;
-}
-#else
-static inline void slab_set_stride(struct slab *slab, unsigned short stride)
-{
-	VM_WARN_ON_ONCE(stride != sizeof(struct slabobj_ext));
-}
-static inline unsigned short slab_get_stride(struct slab *slab)
-{
-	return sizeof(struct slabobj_ext);
-}
-#endif
-
-/*
- * slab_obj_ext - get the pointer to the slab object extension metadata
- * associated with an object in a slab.
- * @slab: a pointer to the slab struct
- * @obj_exts: a pointer to the object extension vector
- * @index: an index of the object
- *
- * Returns a pointer to the object extension associated with the object.
- * Must be called within a section covered by get/put_slab_obj_exts().
- */
-static inline struct slabobj_ext *slab_obj_ext(struct slab *slab,
-					       unsigned long obj_exts,
-					       unsigned int index)
-{
-	struct slabobj_ext *obj_ext;
-
-	VM_WARN_ON_ONCE(obj_exts != slab_obj_exts(slab));
-
-	obj_ext = (struct slabobj_ext *)(obj_exts +
-					 slab_get_stride(slab) * index);
-	return kasan_reset_tag(obj_ext);
+	return (struct slabobj_ext *)(obj_exts & ~OBJEXTS_FLAGS_MASK);
@@ -631,8 +536 @@
-static inline unsigned long slab_obj_exts(struct slab *slab)
-{
-	return 0;
-}
-
-static inline struct slabobj_ext *slab_obj_ext(struct slab *slab,
-					       unsigned long obj_exts,
-					       unsigned int index)
+static inline struct slabobj_ext *slab_obj_exts(struct slab *slab)
@@ -642,4 +539,0 @@
-
-static inline void slab_set_stride(struct slab *slab, unsigned int stride) { }
-static inline unsigned int slab_get_stride(struct slab *slab) { return 0; }
-
@@ -659 +553 @@
-			    void **p, int objects, unsigned long obj_exts);
+			    void **p, int objects, struct slabobj_ext *obj_exts);
