From ff341ed106a05cf730ecdb9950b8e14ee84ad7f7 Mon Sep 17 00:00:00 2001
From: Bernd Waibel <waebbl@gmail.com>
Date: Wed, 1 Jan 2020 11:34:20 +0100
Subject: [PATCH] Add a missing #include directive to find QStringList

Signed-off-by: Bernd Waibel <waebbl@gmail.com>
---
 ApiExtractor/clangparser/clangutils.h | 1 +
 1 file changed, 1 insertion(+)

diff --git a/ApiExtractor/clangparser/clangutils.h b/ApiExtractor/clangparser/clangutils.h
index 98d0c97..01b5a41 100644
--- a/ApiExtractor/clangparser/clangutils.h
+++ b/ApiExtractor/clangparser/clangutils.h
@@ -32,6 +32,7 @@
 #include <clang-c/Index.h>
 #include <QtCore/QPair>
 #include <QtCore/QString>
+#include <QtCore/QStringList>
 #include <QtCore/QVector>
 
 QT_FORWARD_DECLARE_CLASS(QDebug)
-- 
2.24.1

