diff --git a/ApiExtractor/clangparser/compilersupport.cpp b/ApiExtractor/clangparser/compilersupport.cpp
index 74cad05..c641c64 100644
--- a/ApiExtractor/clangparser/compilersupport.cpp
+++ b/ApiExtractor/clangparser/compilersupport.cpp
@@ -239,36 +239,14 @@ static QByteArray noStandardIncludeOption() { return QByteArrayLiteral("-nostdin
 #endif
 
 #if NEED_CLANG_BUILTIN_INCLUDES
-static QString findClang()
-{
-    for (const char *envVar : {"LLVM_INSTALL_DIR", "CLANG_INSTALL_DIR"}) {
-        if (qEnvironmentVariableIsSet(envVar)) {
-            const QString path = QFile::decodeName(qgetenv(envVar));
-            if (QFileInfo::exists(path))
-                return path;
-        }
-    }
-    const QString llvmConfig =
-        QStandardPaths::findExecutable(QLatin1String("llvm-config"));
-    if (!llvmConfig.isEmpty()) {
-        QByteArray stdOut;
-        if (runProcess(llvmConfig, QStringList{QLatin1String("--prefix")}, &stdOut)) {
-            const QString path = QFile::decodeName(stdOut.trimmed());
-            if (QFileInfo::exists(path))
-                return path;
-        }
-    }
-    return QString();
-}
-
 static QString findClangBuiltInIncludesDir()
 {
     // Find the include directory of the highest version.
-    const QString clangPath = findClang();
+    const QString clangPath = QLatin1String("/usr/lib/clang");
     if (!clangPath.isEmpty()) {
         QString candidate;
         QVersionNumber lastVersionNumber(1, 0, 0);
-        QDir clangDir(clangPath + QLatin1String("/lib/clang"));
+        QDir clangDir(clangPath);
         const QFileInfoList versionDirs =
             clangDir.entryInfoList(QDir::Dirs | QDir::NoDotAndDotDot);
         for (const QFileInfo &fi : versionDirs) {
