diff -urN octave-3.2.4/src/DLD-FUNCTIONS/fltk_backend.cc octave-3.2.4_new/src/DLD-FUNCTIONS/fltk_backend.cc
--- octave-3.2.4/src/DLD-FUNCTIONS/fltk_backend.cc	2009-05-25 10:05:00.000000000 +0400
+++ octave-3.2.4_new/src/DLD-FUNCTIONS/fltk_backend.cc	2010-06-30 12:23:38.000000000 +0400
@@ -40,6 +40,14 @@
 #include <sstream>
 #include <iostream>
 
+#include "cmd-edit.h"
+#include "defun-dld.h"
+#include "error.h"
+#include "gl-render.h"
+#include "graphics.h"
+#include "parse.h"
+#include "variables.h"
+
 #include <FL/Fl.H>
 #include <FL/Fl_Window.H>
 #include <FL/Fl_Output.H>
@@ -54,13 +62,6 @@
 #undef max
 #endif
 
-#include "cmd-edit.h"
-#include "defun-dld.h"
-#include "error.h"
-#include "gl-render.h"
-#include "graphics.h"
-#include "parse.h"
-#include "variables.h"
 
 #define FLTK_BACKEND_NAME "fltk"
 
