--- configure.in	2010-01-19 11:13:14.000000000 +0000
+++ configure.in.new	2010-05-05 00:56:31.000000000 +0100
@@ -749,6 +749,11 @@
 GRAPHICS_CFLAGS=
 
 ## fltk (www.fltk.org)
+AC_ARG_WITH(fltk,
+  [AS_HELP_STRING([--without-fltk], [don't use FLTK for native grphics])],
+  with_fltk=$withval, with_fltk=yes)
+
+if test "$with_fltk" = yes; then
 AC_ARG_WITH(fltk-prefix,
         [  --with-fltk-prefix=PFX   Prefix where FLTK is installed (optional)],
         fltk_prefix="$withval",
@@ -773,6 +778,9 @@
 fi
 
 AC_PATH_PROG(FLTK_CONFIG, fltk-config, no)
+else
+   FLTK_CONFIG=no
+fi
 
 warn_fltk_config=""
 warn_fltk_opengl=""
