diff -up octave-3.4.0/src/DLD-FUNCTIONS/__fltk_uigetfile__.cc.fltk octave-3.4.0/src/DLD-FUNCTIONS/__fltk_uigetfile__.cc
--- octave-3.4.0/src/DLD-FUNCTIONS/__fltk_uigetfile__.cc.fltk	2011-02-08 03:00:52.000000000 -0700
+++ octave-3.4.0/src/DLD-FUNCTIONS/__fltk_uigetfile__.cc	2011-05-26 14:09:51.277484258 -0600
@@ -27,7 +27,7 @@ along with Octave; see the file COPYING.
 #if defined (HAVE_FLTK)
 
 #include <FL/Fl.H>
-#include <Fl/Fl_File_Chooser.H>
+#include <FL/Fl_File_Chooser.H>
 
 // FLTK headers may include X11/X.h which defines Complex, and that
 // conflicts with Octave's Complex typedef.  We don't need the X11
diff -up octave-3.4.0/src/DLD-FUNCTIONS/__init_fltk__.cc.fltk octave-3.4.0/src/DLD-FUNCTIONS/__init_fltk__.cc
--- octave-3.4.0/src/DLD-FUNCTIONS/__init_fltk__.cc.fltk	2011-05-26 14:10:40.447667595 -0600
+++ octave-3.4.0/src/DLD-FUNCTIONS/__init_fltk__.cc	2011-05-26 14:10:14.456627808 -0600
@@ -44,7 +44,7 @@ To initialize:
 #include <FL/Fl_Box.H>
 #include <FL/Fl_Button.H>
 #include <FL/Fl_Choice.H>
-#include <Fl/Fl_File_Chooser.H>
+#include <FL/Fl_File_Chooser.H>
 #include <FL/Fl_Gl_Window.H>
 #include <FL/Fl_Menu_Bar.H>
 #include <FL/Fl_Menu_Button.H>
