--- src/liboslexec/gabornoise.cpp	2012-11-17 10:29:28.309215219 +0600
+++ src/liboslexec/gabornoise.cpp	2012-11-17 10:31:11.226266498 +0600
@@ -303,7 +303,7 @@
                 multMatrix (gp.local, x_k_i, xkit);
                 Dual2<Vec2> x_k_i_t = make_Vec2 (comp(xkit,0), comp(xkit,1));
                 Dual2<float> gk = gabor_kernel (w_i_t_s_f, omega_i_t_s_f, phi_i_t_s_f, a_i_t_s_f, x_k_i_t); // 2D
-                if (! isfinite(gk.val())) {
+                if (! std::isfinite(gk.val())) {
                     // Numeric failure of the filtered version.  Fall
                     // back on the unfiltered.
                     gk = gabor_kernel (gp.weight, omega_i, phi_i, gp.a, x_k_i);  // 3D

