--- orig/src/liboslexec/llvm_headers.h	2013-06-20 03:00:47.000000000 +0400
+++ fixed/src/liboslexec/llvm_headers.h	2013-06-20 02:59:10.000000000 +0400
@@ -34,29 +34,21 @@
 #endif
 
 #include <llvm/Bitcode/ReaderWriter.h>
-#include <llvm/Constants.h>
-#include <llvm/DerivedTypes.h>
+#include <llvm/IR/Constants.h>
+#include <llvm/IR/DerivedTypes.h>
 #include <llvm/ExecutionEngine/GenericValue.h>
 #include <llvm/ExecutionEngine/JIT.h>
 #include <llvm/ExecutionEngine/JITMemoryManager.h>
-#include <llvm/Instructions.h>
-#include <llvm/Intrinsics.h>
+#include <llvm/IR/Instructions.h>
+#include <llvm/IR/Intrinsics.h>
 #include <llvm/Linker.h>
-#include <llvm/LLVMContext.h>
-#include <llvm/Module.h>
+#include <llvm/IR/LLVMContext.h>
+#include <llvm/IR/Module.h>
 #include <llvm/PassManager.h>
-#if OSL_LLVM_VERSION >= 32
-#include <llvm/IRBuilder.h>
-#else
-#include <llvm/Support/IRBuilder.h>
-#endif
+#include <llvm/IR/IRBuilder.h>
 #include <llvm/Support/ManagedStatic.h>
 #include <llvm/Support/MemoryBuffer.h>
 #include <llvm/Support/raw_ostream.h>
-#if OSL_LLVM_VERSION >= 32
-#include <llvm/DataLayout.h>
-#else
-#include <llvm/Target/TargetData.h>
-#endif
+#include <llvm/IR/DataLayout.h>
 
 #endif /* OSL_LLVM_HEADERS_H */
