--- src/liboslcomp/oslcomp.cpp
+++ src/liboslcomp/oslcomp.cpp
@@ -421,6 +421,8 @@
             if (m_output_filename.size() == 0)
                 m_output_filename = default_output_filename ();
 
+            std::cout << "default_output_filename " << default_output_filename() << "\n";
+
             std::ofstream oso_output (m_output_filename.c_str());
             if (! oso_output.good()) {
                 error (ustring(), 0, "Could not open \"%s\"",
@@ -587,6 +589,7 @@
 std::string
 OSLCompilerImpl::default_output_filename ()
 {
+    std::cout << "shadername " << shader_decl()->shadername().c_str() << "\n";
     if (m_shader && shader_decl())
         return shader_decl()->shadername().string() + ".oso";
     return std::string();
