--- src/liboslexec/shadingsys.cpp
+++ src/liboslexec/shadingsys.cpp
@@ -26,6 +26,8 @@
 OF THIS SOFTWARE, EVEN IF ADVISED OF THE POSSIBILITY OF SUCH DAMAGE.
 */
 
+#include <OpenImageIO/fmath.h>
+
 #include <vector>
 #include <string>
 #include <cstdio>
@@ -46,7 +48,6 @@
 #include <OpenImageIO/timer.h>
 #include <OpenImageIO/filesystem.h>
 #include <OpenImageIO/optparser.h>
-#include <OpenImageIO/fmath.h>
 
 using namespace OSL;
 using namespace OSL::pvt;
@@ -1495,9 +1496,6 @@
     out << "        Instance param values: " << m_stat_mem_inst_paramvals.memstat() << '\n';
     out << "        Instance connections:  " << m_stat_mem_inst_connections.memstat() << '\n';
 
-    size_t jitmem = LLVM_Util::total_jit_memory_held();
-    out << "    LLVM JIT memory: " << Strutil::memformat(jitmem) << '\n';
-
     if (m_profile) {
         out << "  Execution profile:\n";
         out << "    Total shader execution time: "
