--- src/liboslexec/CMakeLists.txt
+++ src/liboslexec/CMakeLists.txt
@@ -152,11 +152,11 @@
 
 # Unit tests
 add_executable (accum_test accum_test.cpp)
-target_link_libraries ( accum_test oslexec oslcomp ${Boost_LIBRARIES} ${CMAKE_DL_LIBS} ${LLVM_LIBRARY} ${EXTRA_OSLEXEC_LIBRARIES} )
+target_link_libraries ( accum_test oslexec oslcomp ${Boost_LIBRARIES} ${CMAKE_DL_LIBS} ${LLVM_LIBRARY} ${LLVM_MCJIT_LIBRARY} ${EXTRA_OSLEXEC_LIBRARIES} )
 link_ilmbase (accum_test)
 add_test (unit_accum "${CMAKE_BINARY_DIR}/src/liboslexec/accum_test")
 
 add_executable (llvmutil_test llvmutil_test.cpp)
-target_link_libraries ( llvmutil_test oslexec oslcomp ${Boost_LIBRARIES} ${CMAKE_DL_LIBS} ${LLVM_LIBRARY} ${EXTRA_OSLEXEC_LIBRARIES} )
+target_link_libraries ( llvmutil_test oslexec oslcomp ${Boost_LIBRARIES} ${CMAKE_DL_LIBS} ${LLVM_LIBRARY} ${LLVM_MCJIT_LIBRARY} ${EXTRA_OSLEXEC_LIBRARIES} )
 link_ilmbase (llvmutil_test)
 add_test (unit_llvmutil "${CMAKE_BINARY_DIR}/src/liboslexec/llvmutil_test")
