--- a/iconf_pgplot	2008-03-24 14:56:49.000000000 -0000
+++ b/iconf_pgplot	2020-08-16 00:24:25.441414000 -0000
@@ -9,7 +9,7 @@
 #  look for library file
 
 # where to look for pgplot directory: last one found will be used!
-pgdirs="/usr/local/share/ifeffit/pgplot /usr/share/ifeffit/pgplot /Applications/Ifeffit/pgplot $1"
+pgdirs="/usr/local/share/ifeffit/pgplot /usr/share/ifeffit/pgplot /Applications/Ifeffit/pgplot $1 /usr/lib64/pgplot"
 xdirs="/usr/X11/lib /usr/lib/X11 /usr/lib/X11R5 /usr/X11R5/lib /usr/lib/X11R6 /usr/X11R6/lib"
 xd=""
 
