diff --git a/CMakeLists.txt.orig b/CMakeLists.txt
index 9a0e9db..c0fad93 100644
--- a/CMakeLists.txt.orig
+++ b/CMakeLists.txt
@@ -11,6 +11,7 @@ project(cryfs)
 
 list(APPEND CMAKE_MODULE_PATH ${CMAKE_CURRENT_LIST_DIR}/cmake-utils)
 include(utils)
+include(FeatureSummary)
 
 require_gcc_version(7.0)
 require_clang_version(7.0)
@@ -44,10 +45,16 @@ if(MSVC)
   add_definitions(/bigobj)
 endif()
 
+if(BUILD_TESTING)
+    find_package(GTest CONFIG REQUIRED)
+    set(GOOGLETEST_LIBS GTest::gtest GTest::gmock)
+endif()
+
 include(${DEPENDENCY_CONFIG})
 
-add_subdirectory(vendor EXCLUDE_FROM_ALL)
 add_subdirectory(src)
 add_subdirectory(doc)
 add_subdirectory(test)
 add_subdirectory(cpack)
+
+feature_summary(WHAT ALL FATAL_ON_MISSING_REQUIRED_PACKAGES)
diff --git a/src/blockstore/implementations/compressing/compressors/Gzip.cpp.orig b/src/blockstore/implementations/compressing/compressors/Gzip.cpp
index 5420ebf..5f169cf 100644
--- a/src/blockstore/implementations/compressing/compressors/Gzip.cpp.orig
+++ b/src/blockstore/implementations/compressing/compressors/Gzip.cpp
@@ -1,5 +1,5 @@
 #include "Gzip.h"
-#include <vendor_cryptopp/gzip.h>
+#include <cryptopp/gzip.h>
 
 using cpputils::Data;
 
diff --git a/src/cpp-utils/crypto/hash/Hash.cpp.orig b/src/cpp-utils/crypto/hash/Hash.cpp
index 696cdea..e07d28d 100644
--- a/src/cpp-utils/crypto/hash/Hash.cpp.orig
+++ b/src/cpp-utils/crypto/hash/Hash.cpp
@@ -1,6 +1,6 @@
 #include "Hash.h"
 #include <cpp-utils/random/Random.h>
-#include <vendor_cryptopp/sha.h>
+#include <cryptopp/sha.h>
 
 using cpputils::Random;
 using CryptoPP::SHA512;
diff --git a/src/cpp-utils/crypto/kdf/Scrypt.cpp.orig b/src/cpp-utils/crypto/kdf/Scrypt.cpp
index 1fec96c..a12b636 100644
--- a/src/cpp-utils/crypto/kdf/Scrypt.cpp.orig
+++ b/src/cpp-utils/crypto/kdf/Scrypt.cpp
@@ -1,5 +1,5 @@
 #include "Scrypt.h"
-#include <vendor_cryptopp/scrypt.h>
+#include <cryptopp/scrypt.h>
 
 using std::string;
 
diff --git a/src/cpp-utils/crypto/symmetric/CFB_Cipher.h.orig b/src/cpp-utils/crypto/symmetric/CFB_Cipher.h
index c1a8aa1..c0e3d8c 100644
--- a/src/cpp-utils/crypto/symmetric/CFB_Cipher.h.orig
+++ b/src/cpp-utils/crypto/symmetric/CFB_Cipher.h
@@ -6,7 +6,7 @@
 #include "../../data/Data.h"
 #include "../../random/Random.h"
 #include <boost/optional.hpp>
-#include <vendor_cryptopp/modes.h>
+#include <cryptopp/modes.h>
 #include "Cipher.h"
 #include "EncryptionKey.h"
 
diff --git a/src/cpp-utils/crypto/symmetric/GCM_Cipher.h.orig b/src/cpp-utils/crypto/symmetric/GCM_Cipher.h
index 9cf6d53..86b2b8e 100644
--- a/src/cpp-utils/crypto/symmetric/GCM_Cipher.h.orig
+++ b/src/cpp-utils/crypto/symmetric/GCM_Cipher.h
@@ -3,7 +3,7 @@
 #define MESSMER_CPPUTILS_CRYPTO_SYMMETRIC_GCMCIPHER_H_
 
 #include "AEAD_Cipher.h"
-#include <vendor_cryptopp/gcm.h>
+#include <cryptopp/gcm.h>
 
 namespace cpputils {
 
diff --git a/src/cpp-utils/crypto/symmetric/ciphers.h.orig b/src/cpp-utils/crypto/symmetric/ciphers.h
index 0bae686..eee3111 100644
--- a/src/cpp-utils/crypto/symmetric/ciphers.h.orig
+++ b/src/cpp-utils/crypto/symmetric/ciphers.h
@@ -2,12 +2,12 @@
 #ifndef MESSMER_CPPUTILS_CRYPTO_SYMMETRIC_CIPHERS_H_
 #define MESSMER_CPPUTILS_CRYPTO_SYMMETRIC_CIPHERS_H_
 
-#include <vendor_cryptopp/aes.h>
-#include <vendor_cryptopp/twofish.h>
-#include <vendor_cryptopp/serpent.h>
-#include <vendor_cryptopp/cast.h>
-#include <vendor_cryptopp/mars.h>
-#include <vendor_cryptopp/chachapoly.h>
+#include <cryptopp/aes.h>
+#include <cryptopp/twofish.h>
+#include <cryptopp/serpent.h>
+#include <cryptopp/cast.h>
+#include <cryptopp/mars.h>
+#include <cryptopp/chachapoly.h>
 #include "GCM_Cipher.h"
 #include "CFB_Cipher.h"
 
diff --git a/src/cpp-utils/data/Data.cpp.orig b/src/cpp-utils/data/Data.cpp
index be94cdb..e283f0e 100644
--- a/src/cpp-utils/data/Data.cpp.orig
+++ b/src/cpp-utils/data/Data.cpp
@@ -1,6 +1,6 @@
 #include "Data.h"
 #include <stdexcept>
-#include <vendor_cryptopp/hex.h>
+#include <cryptopp/hex.h>
 
 using std::istream;
 using std::ofstream;
diff --git a/src/cpp-utils/data/FixedSizeData.h.orig b/src/cpp-utils/data/FixedSizeData.h
index 5883399..17891c9 100644
--- a/src/cpp-utils/data/FixedSizeData.h.orig
+++ b/src/cpp-utils/data/FixedSizeData.h
@@ -2,7 +2,7 @@
 #ifndef MESSMER_CPPUTILS_DATA_FIXEDSIZEDATA_H_
 #define MESSMER_CPPUTILS_DATA_FIXEDSIZEDATA_H_
 
-#include <vendor_cryptopp/hex.h>
+#include <cryptopp/hex.h>
 #include <string>
 #include <array>
 #include <cstring>
diff --git a/src/cpp-utils/random/OSRandomGenerator.h.orig b/src/cpp-utils/random/OSRandomGenerator.h
index f522c61..d99f977 100644
--- a/src/cpp-utils/random/OSRandomGenerator.h.orig
+++ b/src/cpp-utils/random/OSRandomGenerator.h
@@ -3,7 +3,7 @@
 #define MESSMER_CPPUTILS_RANDOM_OSRANDOMGENERATOR_H
 
 #include "RandomGenerator.h"
-#include <vendor_cryptopp/osrng.h>
+#include <cryptopp/osrng.h>
 
 namespace cpputils {
     class OSRandomGenerator final : public RandomGenerator {
diff --git a/src/cpp-utils/random/RandomGeneratorThread.h.orig b/src/cpp-utils/random/RandomGeneratorThread.h
index 593750e..103c00d 100644
--- a/src/cpp-utils/random/RandomGeneratorThread.h.orig
+++ b/src/cpp-utils/random/RandomGeneratorThread.h
@@ -4,7 +4,7 @@
 
 #include "../thread/LoopThread.h"
 #include "ThreadsafeRandomDataBuffer.h"
-#include <vendor_cryptopp/osrng.h>
+#include <cryptopp/osrng.h>
 
 namespace cpputils {
     //TODO Test
diff --git a/src/cryfs/impl/localstate/BasedirMetadata.cpp.orig b/src/cryfs/impl/localstate/BasedirMetadata.cpp
index d32ced9..3de2d3a 100644
--- a/src/cryfs/impl/localstate/BasedirMetadata.cpp.orig
+++ b/src/cryfs/impl/localstate/BasedirMetadata.cpp
@@ -1,7 +1,7 @@
 #include "BasedirMetadata.h"
 #include <boost/property_tree/ptree.hpp>
 #include <boost/property_tree/json_parser.hpp>
-#include <vendor_cryptopp/sha.h>
+#include <cryptopp/sha.h>
 #include <boost/filesystem/operations.hpp>
 #include "LocalStateDir.h"
 #include <cpp-utils/logging/logging.h>
diff --git a/test/blobstore/CMakeLists.txt.orig b/test/blobstore/CMakeLists.txt
index 05e98b8..342d562 100644
--- a/test/blobstore/CMakeLists.txt.orig
+++ b/test/blobstore/CMakeLists.txt
@@ -27,7 +27,7 @@ set(SOURCES
 )
 
 add_executable(${PROJECT_NAME} ${SOURCES})
-target_link_libraries(${PROJECT_NAME} my-gtest-main googletest blobstore)
+target_link_libraries(${PROJECT_NAME} my-gtest-main ${GOOGLETEST_LIBS} blobstore)
 add_test(${PROJECT_NAME} ${PROJECT_NAME})
 
 target_enable_style_warnings(${PROJECT_NAME})
diff --git a/test/blockstore/CMakeLists.txt.orig b/test/blockstore/CMakeLists.txt
index ca63acc..6dc5f50 100644
--- a/test/blockstore/CMakeLists.txt.orig
+++ b/test/blockstore/CMakeLists.txt
@@ -42,7 +42,7 @@ set(SOURCES
 )
 
 add_executable(${PROJECT_NAME} ${SOURCES})
-target_link_libraries(${PROJECT_NAME} my-gtest-main googletest blockstore)
+target_link_libraries(${PROJECT_NAME} my-gtest-main ${GOOGLETEST_LIBS} blockstore)
 add_test(${PROJECT_NAME} ${PROJECT_NAME})
 
 target_enable_style_warnings(${PROJECT_NAME})
diff --git a/test/cpp-utils/CMakeLists.txt.orig b/test/cpp-utils/CMakeLists.txt
index a13ad98..6f4fb93 100644
--- a/test/cpp-utils/CMakeLists.txt.orig
+++ b/test/cpp-utils/CMakeLists.txt
@@ -71,7 +71,7 @@ target_activate_cpp14(${PROJECT_NAME}_exit_signal)
 target_link_libraries(${PROJECT_NAME}_exit_signal cpp-utils)
 
 add_executable(${PROJECT_NAME} ${SOURCES})
-target_link_libraries(${PROJECT_NAME} my-gtest-main googletest cpp-utils)
+target_link_libraries(${PROJECT_NAME} my-gtest-main ${GOOGLETEST_LIBS} cpp-utils)
 add_dependencies(${PROJECT_NAME} ${PROJECT_NAME}_exit_status ${PROJECT_NAME}_exit_signal)
 add_test(${PROJECT_NAME} ${PROJECT_NAME})
 
diff --git a/test/cryfs-cli/CMakeLists.txt.orig b/test/cryfs-cli/CMakeLists.txt
index 2d0b38c..76fd75b 100644
--- a/test/cryfs-cli/CMakeLists.txt.orig
+++ b/test/cryfs-cli/CMakeLists.txt
@@ -16,7 +16,7 @@ set(SOURCES
 )
 
 add_executable(${PROJECT_NAME} ${SOURCES})
-target_link_libraries(${PROJECT_NAME} my-gtest-main googletest cryfs-cli cryfs-unmount fspp-fuse)
+target_link_libraries(${PROJECT_NAME} my-gtest-main ${GOOGLETEST_LIBS} cryfs-cli cryfs-unmount fspp-fuse)
 add_test(${PROJECT_NAME} ${PROJECT_NAME})
 
 target_enable_style_warnings(${PROJECT_NAME})
diff --git a/test/cryfs/CMakeLists.txt.orig b/test/cryfs/CMakeLists.txt
index eb13b27..0086e6b 100644
--- a/test/cryfs/CMakeLists.txt.orig
+++ b/test/cryfs/CMakeLists.txt
@@ -24,7 +24,7 @@ set(SOURCES
 )
 
 add_executable(${PROJECT_NAME} ${SOURCES})
-target_link_libraries(${PROJECT_NAME} my-gtest-main googletest cryfs)
+target_link_libraries(${PROJECT_NAME} my-gtest-main ${GOOGLETEST_LIBS} cryfs)
 add_test(${PROJECT_NAME} ${PROJECT_NAME})
 
 target_enable_style_warnings(${PROJECT_NAME})
diff --git a/test/cryfs/impl/config/CompatibilityTest.cpp.orig b/test/cryfs/impl/config/CompatibilityTest.cpp
index 07b931b..a045900 100644
--- a/test/cryfs/impl/config/CompatibilityTest.cpp.orig
+++ b/test/cryfs/impl/config/CompatibilityTest.cpp
@@ -2,7 +2,7 @@
 #include <vector>
 #include <boost/filesystem.hpp>
 #include <cpp-utils/data/Data.h>
-#include <vendor_cryptopp/hex.h>
+#include <cryptopp/hex.h>
 #include <cpp-utils/crypto/symmetric/ciphers.h>
 #include <cpp-utils/tempfile/TempFile.h>
 #include <cryfs/impl/config/CryConfigFile.h>
diff --git a/test/fspp/CMakeLists.txt.orig b/test/fspp/CMakeLists.txt
index c6f2e32..295fe58 100644
--- a/test/fspp/CMakeLists.txt.orig
+++ b/test/fspp/CMakeLists.txt
@@ -103,7 +103,7 @@ set(SOURCES
         testutils/OpenFileHandle.cpp testutils/OpenFileHandle.h)
 
 add_executable(${PROJECT_NAME} ${SOURCES})
-target_link_libraries(${PROJECT_NAME} my-gtest-main googletest fspp-interface fspp-fuse)
+target_link_libraries(${PROJECT_NAME} my-gtest-main ${GOOGLETEST_LIBS} fspp-interface fspp-fuse)
 add_test(${PROJECT_NAME} ${PROJECT_NAME})
 
 target_enable_style_warnings(${PROJECT_NAME})
diff --git a/test/gitversion/CMakeLists.txt.orig b/test/gitversion/CMakeLists.txt
index 51a5ccc..396289f 100644
--- a/test/gitversion/CMakeLists.txt.orig
+++ b/test/gitversion/CMakeLists.txt
@@ -6,7 +6,7 @@ set(SOURCES
 )
 
 add_executable(${PROJECT_NAME} ${SOURCES})
-target_link_libraries(${PROJECT_NAME} my-gtest-main googletest gitversion)
+target_link_libraries(${PROJECT_NAME} my-gtest-main ${GOOGLETEST_LIBS} gitversion)
 add_test(${PROJECT_NAME} ${PROJECT_NAME})
 
 target_enable_style_warnings(${PROJECT_NAME})
diff --git a/test/my-gtest-main/CMakeLists.txt.orig b/test/my-gtest-main/CMakeLists.txt
index 1d1e7e0..de4fd10 100644
--- a/test/my-gtest-main/CMakeLists.txt.orig
+++ b/test/my-gtest-main/CMakeLists.txt
@@ -5,7 +5,7 @@ set(SOURCES
 )
 
 add_library(${PROJECT_NAME} STATIC ${SOURCES})
-target_link_libraries(${PROJECT_NAME} PUBLIC googletest cpp-utils)
+target_link_libraries(${PROJECT_NAME} PUBLIC ${GOOGLETEST_LIBS} cpp-utils)
 target_add_boost(${PROJECT_NAME} filesystem system)
 target_include_directories(${PROJECT_NAME} PUBLIC .)
 
diff --git a/test/parallelaccessstore/CMakeLists.txt.orig b/test/parallelaccessstore/CMakeLists.txt
index 16170d1..9737930 100644
--- a/test/parallelaccessstore/CMakeLists.txt.orig
+++ b/test/parallelaccessstore/CMakeLists.txt
@@ -6,7 +6,7 @@ set(SOURCES
 )
 
 add_executable(${PROJECT_NAME} ${SOURCES})
-target_link_libraries(${PROJECT_NAME} my-gtest-main googletest parallelaccessstore)
+target_link_libraries(${PROJECT_NAME} my-gtest-main ${GOOGLETEST_LIBS} parallelaccessstore)
 add_test(${PROJECT_NAME} ${PROJECT_NAME})
 
 target_enable_style_warnings(${PROJECT_NAME})
