--- a/tilelang/contrib/cc.py
+++ b/tilelang/contrib/cc.py
@@ -77,7 +77,11 @@ def get_cplus_compiler():
         The path to the default C/C++ compiler, or None if none was found.
     """
 
-    env_cxx = os.environ.get("CXX") or os.environ.get("CC")
+    env_cxx = (
+        os.environ.get("CUDAHOSTCXX")
+        or os.environ.get("CXX")
+        or os.environ.get("CC")
+    )
     if env_cxx:
         return env_cxx
 
