--- orig/work/iHateTheCashew/ihtc.cpp	2009-12-17 19:20:27.000000000 +0000
+++ fix/work/iHateTheCashew/ihtc.cpp	2011-05-30 23:45:02.000000000 +0000
@@ -116,7 +116,7 @@
     foreach (QGraphicsItem *kid, desktop->childItems())
     {
         o = dynamic_cast<QObject*>(kid);
-        if (o && o->inherits("Plasma::ToolBox") || o->inherits("Plasma::DesktopToolBox"))
+        if (o && o->inherits("Plasma::ToolBox") || o->inherits("Plasma::DesktopToolBox") || o->inherits("DesktopToolBox"))
             return kid;
     }
     return 0;
