--- PyImathNumpy/imathnumpymodule.cpp
+++ PyImathNumpy/imathnumpymodule.cpp
@@ -38,6 +38,7 @@
 #include <PyImathVec.h>
 #include <iostream>
 #include <boost/format.hpp>
+#define NPY_NO_DEPRECATED_API NPY_1_7_API_VERSION
 #include <numpy/arrayobject.h>
 
 using namespace boost::python;
@@ -115,7 +116,7 @@
     if (PyErr_Occurred()) throw_error_already_set();
     scope().attr("numpy") = numpy;
 
-    import_array();
+    if (_import_array()) { };
 
     scope().attr("__doc__") = "Array wrapping module to overlay imath array data with numpy arrays";
 
