--- pyilmbase-2.0.0/PyImath/Makefile.am.orig	2013-02-01 21:04:54.000000000 +0100
+++ pyilmbase-2.0.0/PyImath/Makefile.am	2013-03-29 11:40:21.833417739 +0100
@@ -80,7 +80,7 @@
     PyImathBasicTypes.cpp 
 
 imathmodule_la_LDFLAGS = -avoid-version -module
-imathmodule_la_LIBADD  = -lPyImath @BOOST_PYTHON_LIBS@
+imathmodule_la_LIBADD  = libPyImath.la @BOOST_PYTHON_LIBS@
 
 noinst_HEADERS = PyImathFun.h \
     PyImathBasicTypes.h 
