--- a/CMakeLists.txt	2021-10-12 00:47:46.000000000 +0200
+++ b/CMakeLists.txt	2026-04-24 18:53:27.212711196 +0200
@@ -1,7 +1,7 @@
 cmake_minimum_required(VERSION 3.5)
 message("CMake prefix path: " ${CMAKE_PREFIX_PATH})
 project(PRISM)
-set(CMAKE_CXX_STANDARD 11)
+set(CMAKE_CXX_STANDARD 17)
 set(CMAKE_BUILD_TYPE RELEASE)
 
 # set warning level
@@ -25,7 +25,7 @@
 
 #set (CMAKE_BUILD_TYPE DEBUG)
 if (PRISMATIC_ENABLE_GPU)
-    set(NVCC_FLAGS_EXTRA "-arch=sm_60 -std=c++11 -Xcompiler -fPIC")
+    set(NVCC_FLAGS_EXTRA "-arch=sm_60 -std=c++17 -Xcompiler -fPIC")
 endif (PRISMATIC_ENABLE_GPU)
 set(CMAKE_MODULE_PATH ${CMAKE_SOURCE_DIR}/cmake/modules)
 
