Index: b/gdk/gdk.symbols
===================================================================
--- a/gdk/gdk.symbols
+++ b/gdk/gdk.symbols
@@ -67,6 +70,9 @@
 gdk_device_grab_info_libgtk_only
 gdk_device_list_axes
 gdk_device_list_slave_devices
+#ifdef GDK_WINDOWING_WAYLAND
+gdk_device_manager_core_get_type
+#endif
 gdk_device_manager_get_client_pointer
 gdk_device_manager_get_display
 gdk_device_manager_get_type
@@ -372,6 +375,28 @@
 gdk_visual_get_type
 gdk_visual_get_visual_type
 gdk_visual_type_get_type
+#ifdef GDK_WINDOWING_WAYLAND
+gdk_wayland_app_launch_context_get_type
+gdk_wayland_device_clear_selection_content_libgtk_only
+gdk_wayland_device_get_selection_type_atoms_libgtk_only
+gdk_wayland_device_get_type
+gdk_wayland_device_get_wl_keyboard
+gdk_wayland_device_get_wl_pointer
+gdk_wayland_device_get_wl_seat
+gdk_wayland_device_offer_selection_content_libgtk_only
+gdk_wayland_device_request_selection_content_libgtk_only
+gdk_wayland_display_broadcast_startup_message
+gdk_wayland_display_get_type
+gdk_wayland_display_get_wl_compositor
+gdk_wayland_display_get_wl_display
+gdk_wayland_display_get_wl_shell
+gdk_wayland_display_manager_get_type
+gdk_wayland_drag_context_get_type
+gdk_wayland_window_get_type
+gdk_wayland_window_get_wl_shell_surface
+gdk_wayland_window_get_wl_surface
+gdk_wayland_window_set_modal_hint
+#endif
 #ifdef GDK_WINDOWING_WIN32
 gdk_win32_handle_table_lookup
 gdk_win32_icon_to_pixbuf_libgtk_only
