Index: gtk+-3.6.4/gtk/gtkapplication.c
===================================================================
--- gtk+-3.6.4.orig/gtk/gtkapplication.c	2013-01-10 16:04:08.029595489 +1300
+++ gtk+-3.6.4/gtk/gtkapplication.c	2013-01-10 16:04:08.021595489 +1300
@@ -47,8 +47,6 @@
 #include <gdk/x11/gdkx.h>
 #endif
 
-extern void _gtk_accessibility_shutdown (void);
-
 /**
  * SECTION:gtkapplication
  * @title: GtkApplication
@@ -425,16 +423,12 @@
   gtk_application_shutdown_quartz (GTK_APPLICATION (application));
 #endif
 
-  /* Keep this section in sync with gtk_main() */
-
   /* Try storing all clipboard data we have */
   _gtk_clipboard_store_all ();
 
   /* Synchronize the recent manager singleton */
   _gtk_recent_manager_sync ();
 
-  _gtk_accessibility_shutdown ();
-
   G_APPLICATION_CLASS (gtk_application_parent_class)
     ->shutdown (application);
 }
Index: gtk+-3.6.4/gtk/gtkmain.c
===================================================================
--- gtk+-3.6.4.orig/gtk/gtkmain.c	2013-01-10 16:04:08.029595489 +1300
+++ gtk+-3.6.4/gtk/gtkmain.c	2013-01-10 16:04:08.025595489 +1300
@@ -1170,15 +1170,11 @@
 
   if (gtk_main_loop_level == 0)
     {
-      /* Keep this section in sync with gtk_application_shutdown() */
-
       /* Try storing all clipboard data we have */
       _gtk_clipboard_store_all ();
 
       /* Synchronize the recent manager singleton */
       _gtk_recent_manager_sync ();
-
-      _gtk_accessibility_shutdown ();
     }
 }
 
