--- a/configure	2014-08-25 00:30:09.000000000 +0300
+++ b/configure	2014-09-07 15:01:08.948292323 +0300
@@ -4601,9 +4601,9 @@
 # TODO: autodetect of XML parser libstrophe linked with
 CFLAGS_RESTORE="$CFLAGS"
 CFLAGS="$CFLAGS $AM_CPPFLAGS"
-{ $as_echo "$as_me:${as_lineno-$LINENO}: checking for parser_new in -lstrophe" >&5
-$as_echo_n "checking for parser_new in -lstrophe... " >&6; }
-if ${ac_cv_lib_strophe_parser_new+:} false; then :
+{ $as_echo "$as_me:${as_lineno-$LINENO}: checking for xmpp_ctx_new in -lstrophe" >&5
+$as_echo_n "checking for xmpp_ctx_new in -lstrophe... " >&6; }
+if ${ac_cv_lib_strophe_xmpp_ctx_new+:} false; then :
   $as_echo_n "(cached) " >&6
 else
   ac_check_lib_save_LIBS=$LIBS
@@ -4617,27 +4617,27 @@
 #ifdef __cplusplus
 extern "C"
 #endif
-char parser_new ();
+char xmpp_ctx_new ();
 int
 main ()
 {
-return parser_new ();
+return xmpp_ctx_new ();
   ;
   return 0;
 }
 _ACEOF
 if ac_fn_c_try_link "$LINENO"; then :
-  ac_cv_lib_strophe_parser_new=yes
+  ac_cv_lib_strophe_xmpp_ctx_new=yes
 else
-  ac_cv_lib_strophe_parser_new=no
+  ac_cv_lib_strophe_xmpp_ctx_new=no
 fi
 rm -f core conftest.err conftest.$ac_objext \
     conftest$ac_exeext conftest.$ac_ext
 LIBS=$ac_check_lib_save_LIBS
 fi
-{ $as_echo "$as_me:${as_lineno-$LINENO}: result: $ac_cv_lib_strophe_parser_new" >&5
-$as_echo "$ac_cv_lib_strophe_parser_new" >&6; }
-if test "x$ac_cv_lib_strophe_parser_new" = xyes; then :
+{ $as_echo "$as_me:${as_lineno-$LINENO}: result: $ac_cv_lib_strophe_xmpp_ctx_new" >&5
+$as_echo "$ac_cv_lib_strophe_xmpp_ctx_new" >&6; }
+if test "x$ac_cv_lib_strophe_xmpp_ctx_new" = xyes; then :
   cat >>confdefs.h <<_ACEOF
 #define HAVE_LIBSTROPHE 1
 _ACEOF
