Index: gtk+3.0-3.1.92/gtk/gtksettings.c
===================================================================
--- gtk+3.0-3.1.92.orig/gtk/gtksettings.c	2011-08-28 19:41:49.000000000 +0200
+++ gtk+3.0-3.1.92/gtk/gtksettings.c	2011-09-21 15:34:42.751475478 +0200
@@ -408,7 +408,7 @@
                                              g_param_spec_string ("gtk-fallback-icon-theme",
                                                                   P_("Fallback Icon Theme Name"),
                                                                   P_("Name of a icon theme to fall back to"),
-                                                                  NULL,
+                                                                  "gnome",
                                                                   GTK_PARAM_READWRITE),
                                              NULL);
   g_assert (result == PROP_FALLBACK_ICON_THEME);
