Index: gtk+2.0-2.21.5/modules/input/imviqr.c
===================================================================
--- gtk+2.0-2.21.5.orig/modules/input/imviqr.c	2010-06-16 17:27:05.000000000 +0200
+++ gtk+2.0-2.21.5/modules/input/imviqr.c	2010-07-22 17:56:01.021052474 +0200
@@ -244,7 +244,7 @@
   N_("Vietnamese (VIQR)"), /* Human readable name */
   GETTEXT_PACKAGE,	   /* Translation domain */
    GTK_LOCALEDIR,	   /* Dir for bindtextdomain (not strictly needed for "gtk+") */
-  "vi"			   /* Languages for which this module is the default */
+  ""			   /* Languages for which this module is the default */
 };
 
 static const GtkIMContextInfo *info_list[] = {
