--- OpenGTL-0.9.18/OpenCTL/OpenCTL/Program.cpp.omv~	2014-05-16 00:05:10.714779422 +0200
+++ OpenGTL-0.9.18/OpenCTL/OpenCTL/Program.cpp	2014-05-16 00:05:17.631499373 +0200
@@ -28,7 +28,7 @@
 // Passes
 #include <llvm/PassManager.h>
 #include <llvm/Analysis/LoopPass.h>
-#include <llvm/Analysis/Verifier.h>
+#include <llvm/IR/Verifier.h>
 #include <llvm/IR/GlobalVariable.h>
 
 // GTLCore
--- OpenGTL-0.9.18/OpenGTL/GTLCore/Debug.cpp.omv~	2014-05-16 00:02:10.196722050 +0200
+++ OpenGTL-0.9.18/OpenGTL/GTLCore/Debug.cpp	2014-05-16 00:02:26.023510649 +0200
@@ -25,6 +25,7 @@
 #include <llvm/IR/Module.h>
 #include <llvm/IR/Value.h>
 #include <llvm/IR/Type.h>
+#include <llvm/Support/FileSystem.h>
 #include <llvm/Support/Path.h>
 #include <llvm/Support/raw_ostream.h>
 
--- OpenGTL-0.9.18/OpenGTL/GTLCore/ModuleData_p.cpp.omv~	2014-05-16 00:03:39.070740073 +0200
+++ OpenGTL-0.9.18/OpenGTL/GTLCore/ModuleData_p.cpp	2014-05-16 00:04:15.351019574 +0200
@@ -24,9 +24,9 @@
 #include <llvm/IR/Module.h>
 #include <llvm/PassManager.h>
 #include <llvm/Target/TargetMachine.h>
-#include <llvm/Analysis/Verifier.h>
+#include <llvm/IR/Verifier.h>
 #include <llvm/Transforms/Utils/Cloning.h>
-#include <llvm/Linker.h>
+#include <llvm/Linker/Linker.h>
 #ifdef LLVM_27_OR_28
 #include <llvm/System/Path.h>
 #else
